### DISCUSSION TITLE
Any data persistence library recommendation?

### DISCUSSION THREAD

**Discussion by anothertempore** [cawa-93/vite-electron-builder#883] [Q&A] [answered]
I've checked these:
- https://github.com/sindresorhus/electron-store
- https://github.com/electron-userland/electron-json-storage
- https://github.com/reZach/secure-electron-store

Seems like 1 and 2 have bad support for `preload` script structure, while I don't like the last one's API design :( 

Any recommendations from you guys? Thanks in advance!

**Comment by cawa-93** [comment, marked as answer]
https://github.com/cawa-93/electron-nano-store — A minimalistic, secure, type-safe data store for Electron 


## Usage

```ts
// In Electron Preload Script
import {defineStore} from 'electron-nano-store';

type UserStore = {
  role: 'admin' | 'user'
}

export const storePromise = defineStore<UserStore>('store-name');
```
```ts
// In Renderer
import { storePromise } from '#preload';

const store = await storePromise
store.get('role') // 'admin' | 'user'
store.set('role', 'wrong-role') // TS Error
```

**Reply by anothertempore** [reply]
Thank you Alex! 

TIL we can use this way to figure out user path: https://github.com/cawa-93/electron-nano-store/blob/main/src/index.ts#L7
